Sources/ArcGISToolkit/Components/FormView/FormViewModel.swift

```diff
@@ -46,27 +46,23 @@ import SwiftUI
     }
     
     deinit {
-        // Cancel all `isVisible` tasks.
-        isVisibleTasks.forEach { task in
-            task.cancel()
-        }
-        isVisibleTasks.removeAll()
-        
-        // Cancel expression evaluation.
         evaluateTask?.cancel()
+        isVisibleTask?.cancel()
     }
     
     /// Kick off tasks to monitor `isVisible` for each element.
-    func initializeIsVisibleTasks() {
-        featureForm.elements.forEach { element in
-            let newTask = Task.detached { [unowned self] in
-                for await _ in element.$isVisible {
-                    await MainActor.run {
-                        self.updateVisibleElements()
+    private func initializeIsVisibleTasks() {
+        isVisibleTask = Task {
```

Comment:
Should this be a detached task? Otherwise the task is implicitly holding onto self strongly, preventing it from the model from going away and the task from being cancelled.

Reply:
@philium Is this better? I had to add an `await` and `self.` in `for element in await self.featureForm.elements`.

Is the use of `[unowned self]` OK or should it be something else?

```
    private func initializeIsVisibleTasks() {
        isVisibleTask = Task.detached { [unowned self] in
            await withTaskGroup(of: Void.self) { [unowned self] group in
                for element in await self.featureForm.elements {
                    group.addTask {
                        for await _ in element.$isVisible {
                            guard !Task.isCancelled else { return }
                            await self.updateVisibleElements()
                        }
                    }
                }
            }
        }
    }
```

Reply:
Using `unowned` might be alright, but it is probably best to use `weak`, just to be on the safe side. Remember that a captures a value as it is, so `[weak self]` is only needed on the outer closure (`Task.detached { [weak self] in ... }`), not on the call to `withTaskGroup`.